src/components/Select/Select.tsx

```diff
@@ -149,10 +152,21 @@ export function Select({
 
   const optionsMarkup = normalizedOptions.map(renderOption);
 
+  let labelMarkup = label;
+
+  if (requiredIndicator) {
+    labelMarkup = (
+      <span
+      className={styles.requiredIndicator}>
+        {label}
+      </span>
+    );
+  }
```

Comment:
Nitpick to avoid mutable variable:

```suggestion
  const labelMarkup = requiredIndicator ? (
    <span className={styles.requiredIndicator}>{label}</span>
  ) : (
    label
  );
```